src/Gateway.php

```diff
@@ -164,6 +164,54 @@ public function updateSubscription(array $parameters = array())
         return $this->createRequest('\Omnipay\Vindicia\Message\CreateSubscriptionRequest', $parameters, true);
     }
 
+    /**
+     * Fetch the invoice number of a subscription
+     *
+     * See Message\FetchSubscriptionInvoiceRequest for more details.
+     *
+     * @param array $parameters
+     * @return \Omnipay\Vindicia\Message\FetchSubscriptionInvoiceRequest
+     */
+    public function fetchSubscriptionInvoiceNumbers(array $parameters = array())
+    {
+        /**
+         * @var \Omnipay\Vindicia\Message\FetchSubscriptionInvoiceRequest
+         */
+        return $this->createRequest('\Omnipay\Vindicia\Message\FetchSubscriptionInvoiceRequest', $parameters);
```

Comment:
the way these two functions are reusing the same request object is a bit confusing and it means that you can't enforce the invoice id is passed when it's supposed to be. this means that if you called fetchSubscriptionInvoiceNumbers with the parameters for fetchSubscriptionInvoice, it would just fetch the invoice, when it would actually be more clear to throw an error. can you split it into two separate requests? one can extend the other if you want

Reply:
I thought about it for a long time. I think you are right we should split this to be separate request objects. I think ideally we should have individual request object for every method call so that it is easier to make error checking and unit testing. Also those objects should not form inheritance  since they are fetching different things and have no parent-child relationship in oop perspective.

Reply:
sounds good to me. 

Omnipay does tend to use inheritance for requests that don't really have a parent-child relationship. For example, purchase requests typically extend authorize requests. I'm not sure I agree with the way they do that, but it is definitely the style of the project, so I think it's fine to use inheritance in this way if you want. Either way works.